stdlib/ephemeron.mli

```diff
@@ -75,43 +75,106 @@ module type S = sig
       Use [filter_map_inplace] in this case.
   *)
 
-  include Hashtbl.S
```

Comment:
Is an analogous restriction of the output signature of `Weak.Make` also necessary?